packages/amplify-graphql-model-transformer/src/resources/iterative-dynamo-model-resource-generator.ts

```diff
@@ -0,0 +1,260 @@
+      // This name is used by the mock functionality. Changing this can break mock.
+      const tableBaseName = ctx.resourceHelper.getModelNameMapping(model!.name.value);
+      const tableLogicalName = ModelResourceIDs.ModelTableResourceID(tableBaseName);
+      const scope = ctx.stackManager.getScopeFor(tableLogicalName, tableBaseName);
+
+      this.createModelTable(scope, model, ctx);
+    });
+
+    this.generateResolvers(ctx);
+  }
+
+  createCustomProviderResource(scope: Construct, context: TransformerContextProvider): void {
+    // Policy that grants access to Create/Update/Delete DynamoDB tables
+    const ddbManagerPolicy = new aws_iam.Policy(scope, 'CreateUpdateDeleteTablesPolicy');
+    ddbManagerPolicy.addStatements(
+      new aws_iam.PolicyStatement({
+        actions: ['dynamodb:CreateTable', 'dynamodb:UpdateTable', 'dynamodb:DeleteTable', 'dynamodb:DescribeTable'],
+        // TODO: have more restricted scope
+        resources: ['*'],
+      }),
+    );
+
+    const lambdaCode = aws_lambda.Code.fromAsset(path.join(__dirname, '..', '..', 'lib', 'resources', 'custom-resource-lambda'));
+
+    // lambda that will handle DDB CFN events
+    const gsiOnEventHandler = new aws_lambda.Function(scope, ResourceConstants.RESOURCES.TableOnEventHandlerLogicalID, {
+      runtime: aws_lambda.Runtime.NODEJS_16_X,
+      code: lambdaCode,
+      handler: 'custom-resource-handler.onEvent',
+      timeout: Duration.minutes(14),
+    });
+
+    // lambda that will poll for provisioning to complete
+    const gsiIsCompleteHandler = new aws_lambda.Function(scope, ResourceConstants.RESOURCES.TableIsCompleteHandlerLogicalID, {
+      runtime: aws_lambda.Runtime.NODEJS_16_X,
+      code: lambdaCode,
+      handler: 'custom-resource-handler.isComplete',
+      timeout: Duration.minutes(14),
+    });
+
+    ddbManagerPolicy.attachToRole(gsiOnEventHandler.role!);
+    ddbManagerPolicy.attachToRole(gsiIsCompleteHandler.role!);
+    const gsiCustomProvider = new custom_resources.Provider(scope, ResourceConstants.RESOURCES.TableCustomProviderLogicalID, {
+      onEventHandler: gsiOnEventHandler,
+      isCompleteHandler: gsiIsCompleteHandler,
+      logRetention: aws_logs.RetentionDays.ONE_MONTH,
+      queryInterval: Duration.seconds(30),
+      totalTimeout: Duration.hours(2),
+    });
+    this.customResourceServiceToken = gsiCustomProvider.serviceToken;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getVTLGenerator(): ModelVTLGenerator {
+    return new DynamoDBModelVTLGenerator();
+  }
+
+  protected createModelTable(scope: Construct, def: ObjectTypeDefinitionNode, context: TransformerContextProvider): void {
+    const modelName = def!.name.value;
+    const tableLogicalName = ModelResourceIDs.ModelTableResourceID(modelName);
+    const tableName = context.resourceHelper.generateTableName(modelName);
+
+    // Add parameters.
+    const readIops = new cdk.CfnParameter(scope, ResourceConstants.PARAMETERS.DynamoDBModelTableReadIOPS, {
```

Comment:
Any opportunity to reuse the code above from `generateResources`?

Reply:
I refactored them to a method `createDynamoParameters` in the parent class